New data-source: azurerm_kubernetes_service_versions (#3382)
terraform-providers/terraform-provider-azurerm: website/docs/d/container_orchestrator_versions.html.markdown

---
layout: "azurerm"

> page_title: "Azure Resource Manager: azurerm_container_orchestrator_versions"

Reviewer: ```suggestion
page_title: "Azure Resource Manager: azurerm_kubernetes_service_versions"
```

> sidebar_current: "docs-azurerm-datasource-container-orchestrator-versions-x"

Reviewer: ```suggestion
sidebar_current: "docs-azurerm-datasource-kubernetes-service-versions-x"
```

description: |-

>   Gets available versions of container orchestrators.

Reviewer: ```suggestion
  Gets the available versions of Kubernetes running in Azure Kubernetes Service.
```

---

> # Data Source: azurerm_container_orchestrator_versions

Reviewer: ```suggestion
# Data Source: azurerm_kubernetes_service_versions
```

> Use this data source to access information about available versions of orchestrators supported by the Azure Container services.

Reviewer: ```suggestion
Use this data source to access information about the available versions of Kubernetes supported by the Azure Kubernetes Service.
```

## Example Usage

```hcl

> data "azurerm_container_orchestrator_versions" "current" {

Reviewer: ```suggestion
data "azurerm_kubernetes_service_versions" "current" {
```

>   location = "westeurope"

Reviewer: in the examples we tend to format this as:

```suggestion
  location = "West Europe"
```

(as we automatically transform this internally)

}

output "versions" {

>   value = "${data.azurerm_container_orchestrator_versions.current.versions}"

Reviewer: ```suggestion
  value = "${data.azurerm_kubernetes_service_versions.current.versions}"
```

}

output "latest_version" {

>   value = "${data.azurerm_container_orchestrator_versions.current.latest_version}"

Reviewer: ```suggestion
  value = "${data.azurerm_kubernetes_service_versions.current.latest_version}"
```

}
```

## Argument Reference

* `location` - (Required) Specifies the location in which to query for versions.

> * `orchestrator_type` - (Optional) The type of orchestrator for which to fetch versions. Right now the only supported value is `Kubernetes`. If omitted, defaults to `Kubernetes`.

Reviewer: this has been removed so:

```suggestion
```

> * `version_prefix` - (Optional) A string prefix used to filter the versions retrieved by the query. When this argument is specified, only the versions which match this prefix are returned.

Reviewer: ```suggestion
* `version_prefix` - (Optional) A prefix filter for the versions of Kubernetes which should be returned; for example `1.x` will return `1.9` to `1.14`, whereas `1.12` will return `1.12.2.
```
